### PR TITLE
P4-2515 some test improvement and refactoring around prices.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Small PR with some tidy ups and test improvements around update prices.

### CODE REVIEW

File: src/test/resources/application-test.yml [hunk 1]
```diff
@@ -15,15 +15,7 @@ spring:
       url: "spring.jpa.database: HSQL"
 
   flyway:
-    enabled: true
```
Comment (matt-crowson-moj) [inline src/test/resources/application-test.yml:18]: This config is not used.  Left in error.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/MaintainSupplierPricingControllerTest.kt [hunk 2]
```diff
@@ -68,4 +70,40 @@ class MaintainSupplierPricingControllerTest(@Autowired private val wac: WebAppli
 
     verify(service).addPriceForSupplier(Supplier.SERCO, fromAgencyId, toAgencyId, Money.valueOf(100.24))
   }
+
+  @Test
+  internal fun `cannot add price with characters for Serco`() {
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/MaintainSupplierPricingControllerTest.kt:75]: Additional tests to cover pricing validation.

File: src/main/resources/templates/add-price.html [hunk 3]
```diff
@@ -58,7 +58,7 @@ <h1 class="govuk-heading-xl mv2" th:inline="text">Add price</h1>
             <button class="govuk-button mv3 mr3" data-module="govuk-button">
                 Confirm and save
             </button>
-            <a href="/dashboard" class="govuk-link">Cancel</a>
+            <a href="/journeys" class="govuk-link">Cancel</a>
```
Comment (matt-crowson-moj) [inline src/main/resources/templates/add-price.html:61]: When you add a price it should go back to the journeys for review page.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/MaintainSupplierPricingController.kt [hunk 4]
```diff
@@ -112,23 +109,16 @@ class MaintainSupplierPricingController(@Autowired val supplierPricingService: S
...
-        }
-        if (to != "") {
-            url.queryParam(HtmlController.DROP_OFF_ATTRIBUTE, to)
-        }
-        return "redirect:${url.build().toUri()}"
+
+        from.takeUnless { it == "" }.apply { url.queryParam(HtmlController.PICK_UP_ATTRIBUTE, from) }
+        to.takeUnless { it == "" }.apply { url.queryParam(HtmlController.DROP_OFF_ATTRIBUTE, to) }
+
+        return RedirectView(url.build().toUriString())
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/MaintainSupplierPricingController.kt:116]: Changed simply to get rid of annoying intellij warnings...

**Review by danielepolencic — APPROVED** [review]

**Pull request merged**